### PR TITLE
Update sass lint

### PR DESCRIPTION
**Pull request by catherine-jones**
- include sass linter
- resolves most errors, some have been switched off as the linter rules need to be updated
- unable to get commenting to ignore rules as referenced in docs would appreciate help

### CODE REVIEW

**Comment by catherine-jones** [comment]
this branch should be merged after fix/high-contrast-html

**Comment by elisechant** [comment]
wow!! 💯

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -43,9 +44,9 @@
     </div>
   </header>
 
-  <div id="main">
+  <main>
```
Comment (elisechant) [inline app/views/layouts/application.html.erb]: Be sure to grep the js for any ref to #main - think there might be one.

File: gulpfile.babel.js [hunk 2]
```diff
@@ -158,11 +158,11 @@ gulp.task('clean:tests', (done) => {
 
 gulp.task('sass', () => {
   return gulp.src(`${DIR_SRC_STYLES}/**/*.scss`)
-  // .pipe(sassLint({ // todo - sass lint
-  // 	configFile: './.sass-lint.yml'
-  // }))
-  // .pipe(sassLint.format())
-  // .pipe(sourcemaps.init())
+  .pipe(sassLint({ // todo - sass lint
```
Comment (elisechant) [inline gulpfile.babel.js]: remove the todo

File: lib/assets/src/styles/_base.scss [hunk 3]
```diff
@@ -35,7 +33,7 @@ h1 {
   font-size: 1.5rem;
   font-weight: 700;
   margin-bottom: 1.5rem;
-  margin-top: $gutter/2;
+  margin-top: $gutter / 2;
```
Comment (elisechant) [inline lib/assets/src/styles/_base.scss]: why is margin top a factor of `$gutter` and margin bottom isn't?

File: lib/assets/src/styles/_layout.scss [hunk 4]
```diff
@@ -23,7 +23,7 @@
   }
 
   @include media($screen-sm) {
-    padding: $spacing-unit/2 0;
+    padding: $spacing-unit / 2 0;
```
Comment (elisechant) [inline lib/assets/src/styles/_layout.scss]: double check that this works properly. you might need to interpolate the Math:

` padding: #{$spacing-unit / 2} 0;`

File: lib/assets/src/styles/dashboard/_card.scss [hunk 5]
```diff
@@ -88,32 +90,46 @@
...
 
-      .widget__title h4 {
-        line-height: 1;
-        font-size: $fallback;
-        font-size: 1rem;
+      .widget__title {
+        h4 {
+          line-height: 1;
+          font-size: $fallback;
+          font-size: 1rem;
```
Comment (elisechant) [inline lib/assets/src/styles/dashboard/_card.scss]: we could make a function for font-size that abstracts the double font-size declaration like you did for gutter() to dry this code up a bit?

File: lib/assets/src/styles/dashboard/_chart.scss [hunk 6]
```diff
@@ -81,16 +119,23 @@ svg:not(:root) {
       text-transform: uppercase;
     }
   }
```
Comment (elisechant) [inline lib/assets/src/styles/dashboard/_chart.scss]: closing brackets are indented too deep

File: lib/assets/src/styles/dashboard/_dashboard.scss [hunk 7]
```diff
@@ -36,22 +36,22 @@
...
   }
 }
 
 .dashboard__kpis {
-  background-color: #fff;
+  background-color: $white;
   margin-bottom: 15px;
 
-  .widget__title:before {
+  .widget__title::before {
```
Comment (elisechant) [inline lib/assets/src/styles/dashboard/_dashboard.scss]: above we have `::before`, here `:before`

File: lib/assets/src/styles/widgets/_widget.scss [hunk 8]
```diff
@@ -248,18 +255,18 @@
     opacity: 0.95;
     display: block;
     position: absolute;
-    padding: $gutter/5 $gutter;
-    background: #fff;
-    box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.3);
+    padding: $gutter / 5 $gutter;
```
Comment (elisechant) [inline lib/assets/src/styles/widgets/_widget.scss]: might have same interpolation issues here

**Comment by elisechant** [comment]
Hey @catherine-jones. I don't see any issue with merging other than high_contrast needs to go first separately. 

Left a couple of comments, otherwise LGTM.

File: lib/assets/src/styles/_layout.scss (hunk 4 above)
Comment (catherine-jones) [inline lib/assets/src/styles/_layout.scss]: my understanding is that Sass cannot perform mathematical operations on strings operations on strings and interpolation creates a string.

File: lib/assets/src/styles/_base.scss (hunk 3 above)
Comment (catherine-jones) [inline lib/assets/src/styles/_base.scss]: no idea

File: lib/assets/src/styles/dashboard/_card.scss (hunk 5 above)
Comment (catherine-jones) [inline lib/assets/src/styles/dashboard/_card.scss]: sound like a great plan

File: lib/assets/src/styles/dashboard/_dashboard.scss (hunk 7 above)
Comment (catherine-jones) [inline lib/assets/src/styles/dashboard/_dashboard.scss]: I am presuming that you meant that we used to have :before and now we have ::before this is to do with the CSS3 spec to differentiate between pseudo elements and pseudo classes.

**Pull request merged**